File: backend/src/main/scala/merchant/api/WithdrawMerchantIncomeApi.scala
package merchant.api

import domain.shared.given

import cats.effect.IO
import domain.merchant.WithdrawMerchantIncomeRequest
import domain.shared.{DeliveryAppState, UserRole}
import merchant.app.*
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val withdrawMerchantIncomeApi: FixedMethodApi0[DeliveryAppState] =
  jsonPostApi0[WithdrawMerchantIncomeRequest, DeliveryAppState](
    routeSegment("api"),
    routeSegment("delivery"),
    routeSegment("merchant-profile"),
    routeSegment("withdraw"),
  )

File: frontend/src/apis/merchant/WithdrawMerchantIncomeApi.ts
import type { DeliveryAppState, WithdrawMerchantIncomeRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { defineJsonPostApi0, routeSegment } from '@/system/api/TypedApiDefinitions'

export const withdrawMerchantIncomeApiDefinition = defineJsonPostApi0<WithdrawMerchantIncomeRequest, DeliveryAppState>([
    routeSegment('api'),
    routeSegment('delivery'),
    routeSegment('merchant-profile'),
    routeSegment('withdraw'),
  ])

export function withdrawMerchantIncome(payload: WithdrawMerchantIncomeRequest) {
  return postNormalizedDeliveryState(withdrawMerchantIncomeApiDefinition, payload)
}
